core/src/main/java/org/geysermc/geyser/registry/mappings/versions/MappingsReader_v1.java

```diff
@@ -0,0 +1,663 @@
+                        GeyserImpl.getInstance().getLogger().error("due to entry: " + entry, e);
+                    }
+                }
+            });
+        }
+    }
+
+    private CustomItemOptions readItemCustomItemOptions(JsonNode node) {
+        CustomItemOptions.Builder customItemOptions = CustomItemOptions.builder();
+
+        JsonNode customModelData = node.get("custom_model_data");
+        if (customModelData != null && customModelData.isInt()) {
+            customItemOptions.customModelData(customModelData.asInt());
+        }
+
+        JsonNode damagePredicate = node.get("damage_predicate");
+        if (damagePredicate != null && damagePredicate.isInt()) {
+            customItemOptions.damagePredicate(damagePredicate.asInt());
+        }
+
+        JsonNode unbreakable = node.get("unbreakable");
+        if (unbreakable != null && unbreakable.isBoolean()) {
+            customItemOptions.unbreakable(unbreakable.asBoolean());
+        }
+
+        JsonNode defaultItem = node.get("default");
+        if (defaultItem != null && defaultItem.isBoolean()) {
+            customItemOptions.defaultItem(defaultItem.asBoolean());
+        }
+
+        return customItemOptions.build();
+    }
+
+    @Override
+    public CustomItemData readItemMappingEntry(JsonNode node) throws InvalidCustomMappingsFileException {
+        if (node == null || !node.isObject()) {
+            throw new InvalidCustomMappingsFileException("Invalid item mappings entry");
+        }
+
+        JsonNode name = node.get("name");
+        if (name == null || !name.isTextual() || name.asText().isEmpty()) {
+            throw new InvalidCustomMappingsFileException("An item entry has no name");
+        }
+
+        CustomItemData.Builder customItemData = CustomItemData.builder()
+                .name(name.asText())
+                .customItemOptions(this.readItemCustomItemOptions(node));
+
+        //The next entries are optional
+        if (node.has("display_name")) {
+            customItemData.displayName(node.get("display_name").asText());
+        }
+
+        if (node.has("icon")) {
+            customItemData.icon(node.get("icon").asText());
+        }
+
+        if (node.has("allow_offhand")) {
+            customItemData.allowOffhand(node.get("allow_offhand").asBoolean());
+        }
+
+        if (node.has("display_handheld")) {
+            customItemData.displayHandheld(node.get("display_handheld").asBoolean());
+        }
+
+        if (node.has("texture_size")) {
+            customItemData.textureSize(node.get("texture_size").asInt());
+        }
+
+        if (node.has("render_offsets")) {
+            JsonNode tmpNode = node.get("render_offsets");
+
+            customItemData.renderOffsets(fromJsonNode(tmpNode));
+        }
+
+        return customItemData.build();
+    }
+
+    /**
+     * Read a block mapping entry from a JSON node and Java identifier
+     * @param identifier The Java identifier of the block
```

Comment:
```suggestion
     * Read a block mapping entry from a JSON node and Java identifier
     *
     * @param identifier The Java identifier of the block
```

Reply:
In fact should this comment not be in the parent class? Unless the override drastically changes input/output

Reply:
https://github.com/GeyserMC/Geyser/pull/3505/commits/aca66be74a3901d2a8d187e2c7d057e0e7f1b9f5#diff-e4e44067c9d73f54a016c65c7beece03ee0df9ba528e3f044cc371aca53f0f21R205